### PR TITLE
ADE Advanced Turrets patch

### PR DESCRIPTION
**Pull request by Theblessedwant**
## Additions

Describe new functionality added by your code, e.g.
- Patched ADE Advanced turrets
## Changes

Describe adjustments to existing features made in this merge, e.g.
-added ADE Advanced turret support

## References

Links to the associated issues or other related pull requests, e.g.
- Closes #[ISSUE_NUMBER]
- Contributes towards #[ISSUE_NUMBER]

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long) 2 hours

### CODE REVIEW

**Comment by Theblessedwant** [comment]
something broke the code, currently trouble shooting

**Comment by Theblessedwant** [comment]
ok code checked and should be working now

File: Patches/ADE Advanced Turrets/Buildings_Security.xml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+					<SightsEfficiency>1.00</SightsEfficiency>
+					<ShotSpread>0.14</ShotSpread>
+					<SwayFactor>2.02</SwayFactor>
+					<Bulk>10.66</Bulk>
+				</statBases>
+				<Properties>
+					<recoilAmount>0.34</recoilAmount>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Bullet_40x46mmGrenade_HE</defaultProjectile>
```
Comment (N7Huntsman) [inline Patches/ADE Advanced Turrets/Buildings_Security.xml:116]: A minor quibble, but 40x53mm would be a bit more suited for this application. It's largely the same projectile, just higher velocity intended to be fired from vehicles.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
A few minor changes to be made.

Additionally, I'm currently getting this error on start-up. `Could not resolve cross-reference to Verse.ResearchProjectDef named HeavyTurrets (wanter=prerequisites)`

File: Patches/ADE Advanced Turrets/Buildings_Security.xml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+					<SwayFactor>1.32</SwayFactor>
+					<Bulk>28.72</Bulk>
+				</statBases>
+				<Properties>
+					<recoilAmount>1.80</recoilAmount>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Bullet_20x128mmOerlikon_FMJ</defaultProjectile>
+					<warmupTime>1.4</warmupTime>
+					<range>91</range>
```
Comment (N7Huntsman) [inline Patches/ADE Advanced Turrets/Buildings_Security.xml:159]: This is probably too long--even the 90mm flak gun only has 86 range. The sniper turret fires a 30mm at 86 range. I think something in the range of 64 - 74 might be appropriate.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

File: Patches/ADE Advanced Turrets/Buildings_Security.xml (hunk 2 above)
Comment (Theblessedwant) [reply Patches/ADE Advanced Turrets/Buildings_Security.xml]: Nerfed turret ranges for advanced and advanced plus turrets

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**